dataconnect/dataconnect/movie-connector/mutations.gql

```diff
@@ -0,0 +1,111 @@
+# Create a movie based on user input
+mutation createMovie(
+  $title: String!
+  $releaseYear: Int!
+  $genre: String!
+  $rating: Float
+  $description: String
+  $imageUrl: String!
+  $tags: [String!] = []
+) @auth(level: USER_EMAIL_VERIFIED) {
+  movie_insert(
+    data: {
+      title: $title
+      releaseYear: $releaseYear
+      genre: $genre
+      rating: $rating
+      description: $description
+      imageUrl: $imageUrl
+      tags: $tags
+    }
+  )
+}
+
+# Update movie information based on the provided ID
+mutation updateMovie(
+  $id: UUID!
+  $title: String
+  $releaseYear: Int
+  $genre: String
+  $rating: Float
+  $description: String
+  $imageUrl: String
+  $tags: [String!] = []
+) @auth(level: USER_EMAIL_VERIFIED) {
+  movie_update(
+    id: $id
+    data: {
+      title: $title
+      releaseYear: $releaseYear
+      genre: $genre
+      rating: $rating
+      description: $description
+      imageUrl: $imageUrl
+      tags: $tags
+    }
+  )
+}
+
+# Delete a movie by its ID
+mutation deleteMovie($id: UUID!) @auth(level: USER_EMAIL_VERIFIED) {
+  movie_delete(id: $id)
+}
+
+# Delete movies with a rating lower than the specified minimum rating
+mutation deleteUnpopularMovies($minRating: Float!) @auth(level: USER_EMAIL_VERIFIED) {
+  movie_deleteMany(where: { rating: { le: $minRating } })
+}
+
+# Add a movie to the user's favorites list
+mutation addFavoritedMovie($movieId: UUID!) @auth(level: USER) {
+  favorite_movie_upsert(data: { userId_expr: "auth.uid", movieId: $movieId })
+}
+
+# Remove a movie from the user's favorites list
+mutation deleteFavoritedMovie($movieId: UUID!) @auth(level: USER) {
+  favorite_movie_delete(key: { userId_expr: "auth.uid", movieId: $movieId })
+}
+
+# Add an actor to the user's favorites list
+mutation addFavoritedActor($actorId: UUID!) @auth(level: USER) {
+  favorite_actor_upsert(data: { userId_expr: "auth.uid", actorId: $actorId })
+}
+
+# Remove an actor from the user's favorites list
+mutation deleteFavoriteActor($actorId: UUID!) @auth(level: USER) {
+  favorite_actor_delete(key: { userId_expr: "auth.uid", actorId: $actorId })
+}
+
+# Add a review for a movie
+mutation addReview(
+  $movieId: UUID!
+  $rating: Int!
+  $reviewText: String!
+) @auth(level: USER) {
+  review_upsert(
+    data: {
+      userId_expr: "auth.uid"
+      movieId: $movieId
+      rating: $rating
+      reviewText: $reviewText
+      reviewDate_date: { today: true }
+    }
+  )
+}
+
+# Delete a user's review for a movie
+mutation deleteReview(
+  $movieId: UUID!
+) @auth(level: USER) {
+  review_delete(key: { userId_expr: "auth.uid", movieId: $movieId })
+}
+
+# Upsert (update or insert) a user based on their username
+mutation upsertUser($username: String!) @auth(level: PUBLIC) {
```

Comment:
`@auth(level: USER)` or `USER_EMAIL_VERIFIED`

`auth.uid` is used below, so it's at least `level: USER`.